Lib/test/test_doctest.py

```diff
@@ -1204,20 +1204,18 @@ def displayhook(): r"""
     ...     3
     ...     '''
     >>> test = doctest.DocTestFinder().find(f)[0]
-    >>> r = doctest.DocTestRunner(verbose=False).run(test)
-    >>> post_displayhook = sys.displayhook
+    >>> with patch('sys.displayhook', sys.__displayhook__):
+    ...     r = doctest.DocTestRunner(verbose=False).run(test)
```

Comment:
Instead of "fixing" the `displayhook` here with `unittest.mock.patch`, the behavior and this workaround should be documented:
* describe that doctest now honors custom `displayhook`s
* add a `.. versionchanged` to indicate the change in behavior
* suggest `unittest.mock.patch` as a way to restore the `displayhook` while running the doctests
* suggest [`testsetup`/`testcleanup`](https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html) as another option

Reply:
@ezio-melotti , I have made the requested changes; please review again.

I'm not so sure which is a right place for the versionchanged stuff.  Right now it's the changed method.  Let me know it it's ok or suggesting patch/testsetup/etc is required (I find this too trivial to add, after all - changing the sys.displayhook is very well documented).

Let me also know if the two-step process with the FutureWarning is required: I'll create another PR with a warning (actual change will be postponed for 3.13, it's correct?).  So far, the current state wasn't documented and seems to be rather a bug then a compatibility breaker...